mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/MailboxOperationLoggingListener.java

```diff
@@ -0,0 +1,68 @@
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra;
+
+import static org.apache.james.mailbox.cassandra.GhostMailbox.MAILBOX_NAME;
+import static org.apache.james.mailbox.cassandra.GhostMailbox.TYPE;
+
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.util.MDCStructuredLogger;
+
+public class MailboxOperationLoggingListener implements MailboxListener {
+    public static final String ADDED = "Added";
+    public static final String REMOVED = "Removed";
+
+    @Override
+    public ListenerType getType() {
+        return ListenerType.ONCE;
+    }
+
+    @Override
+    public ExecutionMode getExecutionMode() {
+        return ExecutionMode.SYNCHRONOUS;
+    }
+
+    @Override
+    public void event(Event event) {
+        if (event instanceof MailboxRenamed) {
+            MailboxRenamed mailboxRenamed = (MailboxRenamed) event;
+            MDCStructuredLogger.forLogger(GhostMailbox.LOGGER)
+                .addField(MAILBOX_NAME, mailboxRenamed.getNewPath())
+                .addField(TYPE, ADDED)
+                .log(logger -> logger.info("Mailbox renamed event"));
+            MDCStructuredLogger.forLogger(GhostMailbox.LOGGER)
+                .addField(MAILBOX_NAME, mailboxRenamed.getMailboxPath())
+                .addField(TYPE, REMOVED)
+                .log(logger -> logger.info("Mailbox renamed event"));
+        }
+        if (event instanceof MailboxDeletion) {
+            MDCStructuredLogger.forLogger(GhostMailbox.LOGGER)
+                .addField(MAILBOX_NAME, event.getMailboxPath())
+                .addField(TYPE, REMOVED)
+                .log(logger -> logger.info("Mailbox deleted event"));
+        }
+        if (event instanceof MailboxAdded) {
+            MDCStructuredLogger.forLogger(GhostMailbox.LOGGER)
+                .addField(MAILBOX_NAME, event.getMailboxPath())
+                .addField(TYPE, ADDED)
```

Comment:
you cannot log the id of the mailbox?

Reply:
Not available with this api. I won't try to add it to it, cf recent failure while doing it.

Reply:
I don't like the "recent failure" argument but if you think the logging is sufficient without it, ok.